Documentation/api-reference.md

```diff
@@ -194,6 +195,145 @@ This API is not recommended for:
 
```

Comment:
Can we use the same **Warning**/**Note** notation everywhere in this doc?

Reply:
Oh Yes!